pkgs/development/libraries/pulsar-client-cpp/default.nix

```diff
@@ -0,0 +1,80 @@
+{ stdenv
+, lib
+, pkgs
+, fetchurl
+, version ? "2.8.1"
+, sha512 ? "23i95k6gs5lh6gw6v6hybv6z8rfzc3mfgj2agb6iicad9ia2sf46wr3rqzi9bdhp0c990vcvn72bkz13i0hnh1hj5gnwqin339a2g0x"
+, boost17x
+, clang
+, llvmPackages
+, cmake
+, curl
+, gcc
+, jsoncpp
+, log4cxx
+, openssl
+, pkgconfig
+, protobuf
+, python37
+, snappy
+, zlib
+, zstd}:
+
+let
+  # Not really sure why I need to do this.. If I call clang-tools without the override it defaults to clang_10
+  clang-tools = pkgs.clang-tools.override {inherit stdenv llvmPackages;};
+in
+  stdenv.mkDerivation rec {
+    pname = "pulsar-client-cpp";
+    inherit version;
+
+    src = fetchurl {
+      inherit sha512;
+      url = "https://archive.apache.org/dist/pulsar/pulsar-${version}/apache-pulsar-${version}-src.tar.gz";
+    };
+
+    sourceRoot = "apache-pulsar-${version}-src/pulsar-client-cpp";
+
+    # python37 used in cmake script to calculate some values
+    # clang-tools needed for clang-format etc
+    nativeBuildInputs = [ cmake python37 pkgconfig ]
+    ++ lib.optional stdenv.isDarwin [ clang clang-tools ]
+    ++ lib.optional stdenv.isLinux [ gcc clang-tools ];
+
+    buildInputs = [ boost17x jsoncpp log4cxx openssl protobuf snappy zstd curl zlib ];
+
+    # since we cant expand $out in cmakeFlags
+    preConfigure = ''cmakeFlags="$cmakeFlags -DCMAKE_INSTALL_LIBDIR=$out/lib"'';
+
+    cmakeFlags = [
+      "-DBUILD_TESTS=OFF"
+      "-DBUILD_PYTHON_WRAPPER=OFF"
+      "-DClangTools_PATH=${clang-tools}/bin"
+    ];
+
+    patches = [ ./reader_configuration_includes.diff ];
+    enableParallelBuilding = true;
+
+  doInstallCheck = true;
+  installCheckPhase = ''
+    echo ${lib.escapeShellArg ''
+      #include <pulsar/Client.h>
+      int main (int argc, char **argv) {
+        pulsar::Client client("pulsar://localhost:6650");
+        return 0;
+      }
+    ''} > test.cc
+        ${if stdenv.isDarwin
+        then "$CC++ test.cc -L $out/lib -I $out/include -lpulsar -o test"
+        else "g++ test.cc -L $out/lib -I $out/include -lpulsar -o test" }
```

Comment:
We can't just set CC to g on darwin?

Reply:
If I don't, it looks for `gcc++`, which is not found